test/e2e/http.go

```diff
@@ -333,3 +333,20 @@ func HasStatusCode(code int) func(*HTTPResponse) bool {
 		return res != nil && res.StatusCode == code
 	}
 }
+
+// ref: Copy and modify defaults from https://golang.org/src/net/http/transport.go
+func makeDisableKeepAlivesTransport() *http.Transport {
+	return &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		DisableKeepAlives:     true,
+	}
```

Comment:
Let's do this so we don't have to worry about keeping in sync (can remove the comment too):
```suggestion
        // nolint:forbidigo
        transport := http.DefaultTransport.(*http.Transport).Clone()
	transport.DisableKeepAlives = true
	return transport
```